pandas/tests/types/test_dtypes.py

```diff
@@ -206,6 +208,162 @@ def test_parser(self):
+                  'period[1D2H]', 'Period[1D2H]', '1D2H']:
+            dt = PeriodDtype(s)
+            self.assertEqual(dt.freq, pd.tseries.offsets.Hour(26))
+            self.assertTrue(is_period_dtype(dt))
+
+    def test_subclass(self):
+        a = PeriodDtype('period[D]')
+        b = PeriodDtype('period[3D]')
+
+        self.assertTrue(issubclass(type(a), type(a)))
+        self.assertTrue(issubclass(type(a), type(b)))
+
+    def test_identity(self):
+        self.assertEqual(PeriodDtype('period[D]'),
+                         PeriodDtype('period[D]'))
+        self.assertIs(PeriodDtype('period[D]'),
+                      PeriodDtype('period[D]'))
+
+        self.assertEqual(PeriodDtype('period[3D]'),
+                         PeriodDtype('period[3D]'))
+        self.assertIs(PeriodDtype('period[3D]'),
+                      PeriodDtype('period[3D]'))
+
+        self.assertEqual(PeriodDtype('period[1S1U]'),
+                         PeriodDtype('period[1000001U]'))
+        self.assertIs(PeriodDtype('period[1S1U]'),
+                      PeriodDtype('period[1000001U]'))
+
+    def test_coerce_to_dtype(self):
+        self.assertEqual(_coerce_to_dtype('period[D]'),
+                         PeriodDtype('period[D]'))
+        self.assertEqual(_coerce_to_dtype('period[3M]'),
+                         PeriodDtype('period[3M]'))
+
+    def test_pandas_dtype(self):
+        self.assertEqual(pandas_dtype('period[D]'),
+                         PeriodDtype('period[D]'))
+        self.assertEqual(pandas_dtype('period[3M]'),
+                         PeriodDtype('period[3M]'))
+        self.assertEqual(pandas_dtype('period[U]'),
+                         PeriodDtype('period[U]'))
+        # capital P
+        self.assertEqual(pandas_dtype('Period[D]'),
+                         PeriodDtype('period[D]'))
+        self.assertEqual(pandas_dtype('Period[3M]'),
+                         PeriodDtype('period[3M]'))
+        self.assertEqual(pandas_dtype('Period[U]'),
+                         PeriodDtype('period[U]'))
+
+        # do not parse freq-like string as period dtype
+        self.assertEqual(pandas_dtype('U'), np.dtype('U'))
+        self.assertEqual(pandas_dtype('S'), np.dtype('S'))
+
+    def test_compat(self):
+        self.assertFalse(is_datetime64_ns_dtype(self.dtype))
+        self.assertFalse(is_datetime64_ns_dtype('period[D]'))
+        self.assertFalse(is_datetime64_dtype(self.dtype))
+        self.assertFalse(is_datetime64_dtype('period[D]'))
+
+    def test_construction_from_string(self):
+        result = PeriodDtype('period[D]')
+        self.assertTrue(is_dtype_equal(self.dtype, result))
+        result = PeriodDtype.construct_from_string('period[D]')
+        self.assertTrue(is_dtype_equal(self.dtype, result))
+        with tm.assertRaises(TypeError):
+            PeriodDtype.construct_from_string('foo')
+        with tm.assertRaises(TypeError):
+            PeriodDtype.construct_from_string('period[foo]')
```

Comment:
add ``datetime64`` and ``datetime64[ns, US/Eastern]`` as additonal tests here (should raise)